app.js

```diff
@@ -0,0 +1,207 @@
+// this is the game board
+const gameBoard = () => {
```

Comment:
Our `gameBoard` factory is a bit overcrowded. I propose that the `gameBoard` 's data and methods contain only the ff:

1. Initial array (the state of the board every start of a game) -> Instead of using a 2-dimensional array, we can simply use 1-D to simplify all the downstream processes involved (i.e, `board = [0, 1, 2, 3, 4, 5, 6, 7, 8,]` instead of `board = [['', '', '']....]`

2. An array representing the winning conditions -> with a 1-dimensional `_board` representation as suggested above, it's going to be simpler checking for winning combinations:

```javascript
const winningCombos = [[0, 1, 2], [3, 4 ,5], ....etc]
```
Assuming your `playerFactory` has an array that stores each player's set of moves you can simply compare those player moves against the winningCombos array to find out whether or not that player has already achieved a winning status.

3. A function that marks the specific index a player has chosen. Each player has either a "X" or "O" symbol. Therefore, whenever a player sends his/her choice of index (again, it's much simpler using a one-dimensional array for your `_board`) this `setMark` method will simply:

```javascript
const setMark = (index, symbol) => {
  typeof board[index] === "number" ? board[index] = symbol : undefined;
  // what this does is first check whether the particular index in the board
 // still has a number as an element, if yes then change it into the 
 // player's symbol, otherwise do nothing
}
```

4. A method that checks whether or not the board is already full. Again the advantage of using a single-dimension array for `_board` here is evident:

```javascript
const isFull = () => _board.every(element => typeof element === "string");

// if all elements are either "X" or "O" then the board is already full -
// the function will then return true, if at least one element 
// is still a number, it will surely return false - meaning the board isn't full yet
```

Think about this suggestion :)

Reply:
We edited our board, fixed from both html and js side so it becomes a one dimensional array.
We also followed your advice on other methods